samples/dfu/README.rst

```diff
@@ -7,44 +7,70 @@ Overview
 This DFU application demonstrates how to connect with Golioth and use Device
 Firmware Upgrade (DFU) procedure.
 
+The app is first built and flashed to the device which will then listen to
+Golioth Cloud for a new firmware binary to become available. A second binary is
+built with a new version number, registered as an artifact, and depolyed as a
+release. The device will detect this change and automatically download and
+install.
+
 Requirements
 ************
 
 - Golioth credentials
 - Network connectivity
 
-Using nRF9160 Feather with nRF Connect SDK
-******************************************
+Using nRF9160 Feather or Thingy:91 with nRF Connect SDK
+*******************************************************
```

Comment:
We do actually support nRF9160 DK as well, so maybe it is good to add it here as well (in separate commit of course).

Reply:
This opens a larger question: do we add build command examples for every supported hardware?

Reply:
Good question. In general `samples/` in this repo should be just samples of what we implement in this repo, which is sample usage of lightdb, stream, dfu, etc. In the beginning we just had 3 platforms that we supported, so it made sense to give overlay files for each platform. Currently this list is much longer and the maintenance effort grows with each added platform, as in theory we should make sure that each commit still works on each platform.